### DISCUSSION TITLE
Does ARC support docker compose calls through Makefile?

### DISCUSSION THREAD

**Discussion by rahul-FLS** [actions/actions-runner-controller#3110] [Questions]
I'm trying to pull down few containers using docker-compose and Makefile by setting `containerMode: kubernetes` but can't seem to get it working.

Error: `Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?`

My understanding was that with `containerMode: kubernetes` set, it will start a new pod to run the job which contains `make` calls. Am I missing something?

**Comment by aladdin-nawasrah** [comment]
Any response here?

**Reply by nobbs** [reply]
That's a misunderstanding what `containerMode: kubernetes` does. This setting only affects the way how [container-based actions](https://docs.github.com/en/actions/creating-actions/creating-a-docker-container-action) and workflows defining a custom `container` section (see [this docs page](https://docs.github.com/en/actions/using-jobs/running-jobs-in-a-container)) are executed - those are run in their own pods in this setup, while without this setting they are run in the docker-in-docker setup in the same pod as the rest of the workflow. 

If you want to run steps or actions that make use of the `docker` cli or `docker compose`, that's not going to work with `containerMode: kubernetes` as they require a proper docker daemon to be running. You can't expect your `docker` or `docker compose` commands to be magically translated to K8s native API calls - that's not how any of this works.

The limitations of the `kubernetes` mode are also listed in the [docs](https://github.com/actions/actions-runner-controller/blob/master/docs/deploying-alternative-runners.md#runner-with-k8s-jobs).
